### PR TITLE
ASAP call oncommit callback function to make OSDOpReply firstly reply.

### PR DESCRIPTION
**Pull request by majianpeng**
Make op_commit->complete ASAP be called because callback of op_commit send reply to client.

### CODE REVIEW

**Comment by liupan1111** [comment]
@majianpeng your first commit has already been pulled by me.

https://github.com/ceph/ceph/pull/13918

**Comment by majianpeng** [comment]
@liupan1111 . Ok, i'll remove this.

**Comment by liupan1111** [comment]
Could you modify your pr title, and make it more meaningful?

File: src/osd/ReplicatedBackend.cc [hunk 1]
```diff
@@ -595,10 +595,10 @@ void ReplicatedBackend::op_applied(
   if (op->waiting_for_applied.empty()) {
     op->on_applied->complete(0);
     op->on_applied = 0;
-  }
-  if (op->done()) {
-    assert(!op->on_commit && !op->on_applied);
-    in_progress_ops.erase(op->tid);
+    if (op->waiting_for_commit.empty()) {
```
Comment (amitkumar50) [inline src/osd/ReplicatedBackend.cc]: @majianpeng tabs left

File: src/osd/ReplicatedBackend.cc [hunk 2]
```diff
@@ -618,10 +618,10 @@ void ReplicatedBackend::op_commit(
   if (op->waiting_for_commit.empty()) {
     op->on_commit->complete(0);
     op->on_commit = 0;
-  }
-  if (op->done()) {
-    assert(!op->on_commit && !op->on_applied);
-    in_progress_ops.erase(op->tid);
+    if (op->waiting_for_applied.empty()) {
+      assert(!op->on_applied);
```
Comment (amitkumar50) [inline src/osd/ReplicatedBackend.cc]: @majianpeng NIT tab left

**Pull request closed without merging**